File: algorithms/cpp/singleNumber/singleNumber.cpp
// Source: https://leetcode.com/problems/single-number/
// Author: Miao Zhang
// Date:   2021-01-21

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int res = 0;
        for (auto num: nums) {
            res ^= num;
        }
        return res;
    }
};

File: algorithms/python/singleNumber/singleNumber.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/single-number/
# Author: Miao Zhang
# Date:   2021-01-21

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        res = 0
        for num in nums:
            res ^= num
        return res
